src/Sylius/Bundle/SettingsBundle/Manager/SettingsManager.php

```diff
@@ -101,20 +102,30 @@ public function __construct(
     /**
      * {@inheritdoc}
      */
-    public function loadSettings($namespace, $ignoreUnknown = true)
+    public function load($schemaAlias, $namespace = null, $ignoreUnknown = true)
     {
-        if (isset($this->resolvedSettings[$namespace])) {
-            return $this->resolvedSettings[$namespace];
+        $schema = $this->schemaRegistry->get($schemaAlias);
+
+        $resolver = $this->defaultResolver;
+
+        if ($this->resolverRegistry->has($schemaAlias)) {
+            $resolver = $this->resolverRegistry->get($schemaAlias);
         }
 
-        if ($this->cache->contains($namespace)) {
-            $parameters = $this->cache->fetch($namespace);
-        } else {
-            $parameters = $this->getParameters($namespace);
-            $this->cache->save($namespace, $parameters);
+        // try to resolve settings for schema alias and namespace
+        $settings = $resolver->resolve($schemaAlias, $namespace);
+
+        if (!$settings) {
+            $settings = $this->settingsFactory->createNew();
+            $settings->setSchemaAlias($schemaAlias);
         }
 
-        $schema = $this->schemaRegistry->getSchema($namespace);
+        // map parameters to a plain php array
+        $parameters = $settings->getParameters()
```

Comment:
I can't come up with an idea for specific stuff that can be done on Parameter level but not on Settings level. It will simplify all the things with introducing just a little bit of restrictions. Overriding `Parameter` also means that the custom stuff is done on each one.

Reducing that complexity we get rid of all Parameter* models, tricky relations and we reduce all the issues related to transforming array <=> ParameterCollection/Parameter.

WDYT? /cc @pjedrzejewski what was behind the inital idea?